python/cudf/cudf/tests/test_datetime.py

```diff
@@ -1334,6 +1336,57 @@ def test_quarter():
     assert_eq(expect2.values, got2.values, check_dtype=False)
 
 
+def test_isocalendar():
```

Comment:
Can you parameterize the test with `Series` & `Index` inputs? There seems to be a lot of duplication of code in this test.

Also, would add a case for `Series` input along with a custom `index`.